lib/neo4j/core/cypher_session/adaptors/http.rb

```diff
@@ -94,12 +95,12 @@ class Requestor
             default_url('http://neo4:neo4j@localhost:7474')
             validate_uri { |uri| uri.is_a?(URI::HTTP) }
 
-            def initialize(url, user_agent_string, instrument_proc)
+            def initialize(url, user_agent_string, instrument_proc, adaptor)
               self.url = url
               @user = user
               @password = password
               @user_agent_string = user_agent_string
-              @faraday = faraday_connection
+              @faraday = faraday_connection(adaptor)
```

Comment:
That way we wouldn't need to include the `typhoeus` gem into our project.  Users of the gem would bring it into the project if they need it.  We could simply test that the options are being passed through while respecting the Faraday API boundary.